assets/js/modules/analytics/datastore/accounts.js

```diff
@@ -276,11 +316,46 @@ const baseResolvers = {
 			dispatch( STORE_NAME ).receiveAccountsPropertiesProfilesCompletion();
 		}
 
-		const accountID = registry.select( STORE_NAME ).getAccountID();
+		let accountID = registry.select( STORE_NAME ).getAccountID();
 		// Pre-select values from the matched property if no account is selected.
 		if ( matchedProperty && ! accountID ) {
-			registry.dispatch( STORE_NAME ).setAccountID( matchedProperty.accountId ); // eslint-disable-line sitekit/acronym-case
-			registry.dispatch( STORE_NAME ).selectProperty( matchedProperty.id, matchedProperty.internalWebPropertyId ); // eslint-disable-line sitekit/acronym-case
+			/* eslint-disable sitekit/acronym-case */
+			accountID = matchedProperty.accountId;
+			registry.dispatch( STORE_NAME ).setAccountID( matchedProperty.accountId );
+			yield Data.commonActions.await( registry.dispatch( STORE_NAME ).selectProperty( matchedProperty.id, matchedProperty.internalWebPropertyId ) );
```

Comment:
`selectProperty` here is an action on the local store so this one can be simplified 😄 
```suggestion
			yield selectProperty( matchedProperty.id, matchedProperty.internalWebPropertyId );
```

Reply:
Updated.